hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -4042,46 +4033,41 @@ else if ( expressible instanceof EmbeddableValuedModelPart ) {
 		}
 		else if ( expressible instanceof EntityValuedModelPart ) {
 			final EntityValuedModelPart entityValuedModelPart = (EntityValuedModelPart) expressible;
-			final Object associationKey;
-			final ModelPart associationKeyPart;
-			if ( entityValuedModelPart instanceof Association ) {
-				final Association association = (Association) entityValuedModelPart;
-				final ForeignKeyDescriptor foreignKeyDescriptor = association.getForeignKeyDescriptor();
-				associationKey = foreignKeyDescriptor.getAssociationKeyFromSide(
-						literal.getLiteralValue(),
-						association.getSideNature(),
-						null
-				);
-				associationKeyPart = foreignKeyDescriptor.getPart( association.getSideNature() );
-			}
-			else {
-				final EntityIdentifierMapping identifierMapping = entityValuedModelPart.getEntityMappingType()
-						.getIdentifierMapping();
-				associationKeyPart = identifierMapping;
-				associationKey = identifierMapping.getIdentifier(
-						literal.getLiteralValue(),
-						null
-				);
-			}
```

Comment:
Note to self, check why removing this is necessary. The LHS of a comparison `cb.equal( root.get( "assoc" ), cb.literal( myEntity ) )` will use the FK part so this change IMO might lead to comparing wrong columns.